ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AvailableStorageTable.tsx

```diff
@@ -154,6 +160,33 @@ const getDiskActions = (
   return actions;
 };
 
+/**
+ * Generate the actions that a given partition can perform.
+ * @param partition - the partition to check.
+ * @param setExpanded - function to set the expanded table row and content.
+ * @returns list of action links.
+ */
+const getPartitionActions = (
+  partition: Partition,
+  setExpanded: (expanded: Expanded | null) => void
+) => {
+  const actions = [];
+  const actionGenerator = (label: string, content: Expanded["content"]) => ({
+    children: label,
+    "data-test": content,
+    onClick: () => setExpanded({ content, id: uniqueId(partition) }),
+  });
```

Comment:
Could this function be shared with `getDiskActions` and also take the value for `uniqueId`?

Reply:
Absolutely!

Reply:
Hmm actually it will require me to add `setExpanded` and `id` to all of the different actions, so it's reducing one bit of duplication but adding a different bit of duplication.

Reply:
I've added [an issue](https://github.com/canonical-web-and-design/maas-ui/issues/2072) to turn the action menus into components which I think will be the best method of reducing duplication. Plus it'll help break down the `AvailableDisksAndPartitions` component which is pretty huge now.